packages/embed-widget/src/App.tsx

```diff
@@ -148,29 +149,57 @@ function App(): JSX.Element {
       }
       initApp();
     },
-    [api, client, connection, dispatch, name, serverConfig, user, shared]
+    [
+      api,
+      client,
+      connection,
+      dispatch,
+      name,
+      serverConfig,
+      user,
+      isShared,
+      type,
+    ]
   );
 
   const isLoaded =
-    (definition != null || sharedObject != null) && error == null;
-  const isLoading = definition == null && sharedObject == null && error == null;
+    (definition != null || sharedReady != false) && error == null;
+  const isLoading = definition == null && sharedReady == false && error == null;
+
+  type ConnectionWithGetSharedObject = dh.IdeConnection & {
+    getSharedObject(name: string, type: string): Promise<unknown>;
+  };
+
+  function isConnectionWithGetSharedObject(
+    connection: dh.IdeConnection
+  ): connection is ConnectionWithGetSharedObject {
+    return (
+      'getSharedObject' in connection &&
+      typeof connection.getSharedObject === 'function'
+    );
+  }
```

Comment:
Move the type definition and typePredicate outside of/above this functional component.